peachjam/models/external_document.py

```diff
@@ -0,0 +1,21 @@
+from django.utils import timezone
+from django.utils.text import slugify
+from languages_plus.models import Language
+
+from .core_document_model import CoreDocument
+
+
+class ExternalDocument(CoreDocument):
+
+    frbr_uri_doctypes = ["doc"]
+
+    def clean(self):
+        self.doc_type = "external"
+        self.frbr_uri_number = slugify(self.title)
```

Comment:
It's better if we do this in the model admin form using `    prepopulated_fields = {"slug": ("title",)}` because then it fills itself in for the user.

Also setting the language and date defaults in the model admin form means the user doesn't have to think about it and leave it blank.

Reply:
Changed